### PR TITLE
Add binary cross entropy loss layer

### PR DESCRIPTION
**Pull request by manifest**
This PR introduces a binary classification loss layer.

It may be used in the supervised training `Loop` as follows:
```python
trax.supervised.training.TrainTask(
    ...
    loss_layer=BinaryCrossEntropyLoss(),
    ...
)
```

Or as an evaluation metric:
```python
trax.supervised.training.EvalTask(
    ...
    metrics=[BinaryCrossEntropyLoss(), Accuracy(classifier=BinaryClassifier())],
)
```

Changes include:
- Additional layers `BinaryClassifier` and `MulticlassClassifier` to separate classification from accuracy estimation. `BinaryClassifier` may be configured with a `threshold` and `MulticlassClassifier` with an `axis`. These layers may also be useful in end-user application code.
- `Accuracy` and `SequenceAccuracy` functions now accept a classification layer argument with `MulticlassClassifier` as a default value to preserve backward compatibility with the current behavior.
- A new `BinaryCrossEntropySum` layer that mimics `CrossEntropySum`.

### CODE REVIEW

**Comment by manifest** [comment]
@googlebot I signed it!

**Comment by manifest** [comment]
Should I rename `CrossEntropy` to `MulticlassCrossEntropy` to make layer names more consistent?

**Comment by afrozenator** [comment]
Thanks @manifest - this looks good, will merge in shortly!
